### PR TITLE
New User Profile Virtual Field Types (Member Since, Last Login) - Ref 4283

### PR DESCRIPTION
**Pull request by MasterHans**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [ ] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [ ] When resolving a specific issue, it's referenced in the PR's description (e.g. `Fix #xxx[,#xxx]`, where "xxx" is the Github issue number)
- [ ] All tests are passing
- [ ] New/updated tests are included
- [ ] Changelog was modified

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Comment by MasterHans** [comment]
@luke-  PR Done!

**Comment by Carl2213** [comment]
Can this be renamed to "Account creation date"? To avoid confusion with "member of space since"

**Comment by luke-** [comment]
I have adjusted the wording a bit. 
It's just the Field Type Description and the label has to be set by the admin anyway.

**Review by luke- — APPROVED** [review]

**Review by luke- — CHANGES_REQUESTED** [review]
Can you please modify the CHANGELOG?

**Comment by MasterHans** [comment]
> Can you please modify the CHANGELOG?

@luke- 
Should I edit **humhub/CHANGELOG.md**   OR  **humhub/CHANGELOG_DEV.md**

**Comment by luke-** [comment]
_DEV

**Comment by MasterHans** [comment]
> _DEV

Done!

**Comment by luke-** [comment]
@MasterHans Did you committed the change?

File: protected/humhub/modules/user/models/fieldtype/UserLastLogin.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public function getVirtualUserValue($user, $raw = true)
+    {
+        if (empty($user->last_login)) {
+            return '';
+        }
+
+        if ($raw) {
+            return Html::encode($user->last_login);
+        }
+    }
```
Comment (luke-) [inline protected/humhub/modules/user/models/fieldtype/UserLastLogin.php:34]: Missing return statement for non raw output. Please use something like `Yii::$app->formatter->asDate`.

image

File: protected/humhub/modules/user/models/fieldtype/UserLastLogin.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+class UserLastLogin extends BaseTypeVirtual
+{
+
+    /**
+     * @inheritDoc
+     */
+    public function getVirtualUserValue($user, $raw = true)
+    {
+        if (empty($user->last_login)) {
+            return '';
```
Comment (luke-) [inline protected/humhub/modules/user/models/fieldtype/UserLastLogin.php:28]: Maybe we should return a "-" here.

File: protected/humhub/modules/user/models/fieldtype/UserMemberSince.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    public function getVirtualUserValue($user, $raw = true)
+    {
+        if (empty($user->created_at)) {
+            return '';
+        }
+
+        if ($raw) {
+            return Html::encode($user->created_at);
+        }
+    }
```
Comment (luke-) [inline protected/humhub/modules/user/models/fieldtype/UserMemberSince.php:34]: Missing return statement

**Review by luke- — CHANGES_REQUESTED** [review]
+ CHANGELOG_DEV.md modification is not commited yet.

File: protected/humhub/modules/user/models/fieldtype/UserLastLogin.php (hunk 1 above)
Comment (MasterHans) [reply protected/humhub/modules/user/models/fieldtype/UserLastLogin.php:34]: Done! @luke-  I did not catch up what is the purpose of the **$raw** parameter? Could you please explain? Thank's.

File: protected/humhub/modules/user/models/fieldtype/UserMemberSince.php (hunk 3 above)
Comment (MasterHans) [reply protected/humhub/modules/user/models/fieldtype/UserMemberSince.php:34]: Done! Sorry for that!

File: protected/humhub/modules/user/models/fieldtype/UserLastLogin.php (hunk 2 above)
Comment (MasterHans) [reply protected/humhub/modules/user/models/fieldtype/UserLastLogin.php]: Agree! Done!

**Review by luke- — APPROVED** [review]

**Pull request merged**